pkgs/servers/mqtt/nanomq/default.nix

```diff
@@ -1,25 +1,65 @@
-{ lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config
-, cyclonedds, libmysqlclient, mariadb, mbedtls, sqlite, zeromq
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+, pkg-config
+, cyclonedds
+, libmysqlclient
+, mariadb
+, mbedtls
+, sqlite
+, zeromq
+, flex
+, bison
+
+# for tests
+, python3
+, mosquitto
+, netcat-gnu
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+let
+
+  # exposing as full package in its own right would be a
+  # bit absurd - repo doesn't even have a license.
+  idl-serial = stdenv.mkDerivation {
+    pname = "idl-serial";
+    version = "unstable-2023-03-29";
+
+    src = fetchFromGitHub {
+      owner = "nanomq";
+      repo = "idl-serial";
+      rev = "908c364dab4c0dcdd77b8de698d29c8a0b6d3830";
+      hash = "sha256-3DS9DuzHN7BevfgiekUmKKH9ej9wKTrt6Fuh427NC4I=";
+    };
+
+    nativeBuildInputs = [ cmake ninja flex bison ];
+  };
+
+in stdenv.mkDerivation (finalAttrs: {
   pname = "nanomq";
-  version = "0.16.3";
+  version = "0.18.1";
 
   src = fetchFromGitHub {
     owner = "emqx";
     repo = "nanomq";
     rev = finalAttrs.version;
-    hash = "sha256-9w4afVxuJbYrkagpAe1diftDnjrRjunyhJdJ0BZq3K0=";
+    hash = "sha256-ld+UuuIWcAIt4XkC+wTbgwNUoIyyyjWi+U90YL3I3Tw=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # https://github.com/emqx/nanomq/pull/1250
```

Comment:
Why not `fetchpatch` https://github.com/emqx/nanomq/pull/1250/commits/ee5fdb37a0f21b238369aef94334f8b8a429647b?

Reply:
Ah - I completely forgot to update this after that was merged.

Standby...